cmd/kubernetesDeploy_test.go

```diff
@@ -454,6 +521,69 @@ spec:
 		assert.Contains(t, string(appTemplate), "my.registry:55555/path/to/Image:latest")
 	})
 
+	t.Run("test kubectl - create secret from docker config.json", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", "")
+		defer os.RemoveAll(dir) // clean up
+		assert.NoError(t, err, "Error when creating temp dir")
+
+		opts := kubernetesDeployOptions{
+			AppTemplate:                filepath.Join(dir, "test.yaml"),
+			DockerConfigJSON:           "/path/to/.docker/config.json",
+			ContainerRegistryURL:       "https://my.registry:55555",
+			ContainerRegistryUser:      "registryUser",
+			ContainerRegistryPassword:  "********",
+			ContainerRegistrySecret:    "regSecret",
+			CreateDockerRegistrySecret: true,
+			DeployTool:                 "kubectl",
+			Image:                      "path/to/Image:latest",
+			AdditionalParameters:       []string{"--testParam", "testValue"},
+			KubeConfig:                 "This is my kubeconfig",
+			KubeContext:                "testCluster",
+			Namespace:                  "deploymentNamespace",
+		}
+
+		kubeYaml := `kind: Deployment
+metadata:
+spec:
+  spec:
+    image: <image-name>`
+
+		ioutil.WriteFile(opts.AppTemplate, []byte(kubeYaml), 0755)
+
+		e := mock.ExecMockRunner{
+			ShouldFailOnCommand: map[string]error{
+				"kubectl --insecure-skip-tls-verify=true --namespace=deploymentNamespace --context=testCluster get secret regSecret": fmt.Errorf("secret not found"),
+			},
+		}
+		var stdout bytes.Buffer
+		runKubernetesDeploy(opts, &e, &stdout)
+
+		assert.Equal(t, e.Env, []string{"KUBECONFIG=This is my kubeconfig"})
+
+		assert.Equal(t, "kubectl", e.Calls[0].Exec, "Wrong secret lookup command")
```

Comment:
The check of e.Calls[0] isn't realy needed, is it?

Reply:
This comes from copying another test ("test kubectl - create secret/kubeconfig"). As it is already tested there we can remove it, yes. Here we will only keep specific asserts for this test case.